jdtls-bin-override/test_jdtls.py

```diff
@@ -0,0 +1,180 @@
+            assert '-Dhttps.proxyPort=8443' in result
+            assert '-Dhttps.proxyUser=user' in result
+            assert '-Dhttps.proxyPassword=pass' in result
+
+    def test_no_proxy_single_host(self):
+        """Test NO_PROXY with a single host."""
+        with patch.dict(os.environ, {'NO_PROXY': 'localhost'}, clear=True):
+            result = get_proxy_jvm_args()
+            assert '-Dhttp.nonProxyHosts=localhost' in result
+
+    def test_no_proxy_multiple_hosts(self):
+        """Test NO_PROXY with comma-separated hosts."""
+        with patch.dict(os.environ, {'NO_PROXY': 'localhost,127.0.0.1,.example.com'}, clear=True):
+            result = get_proxy_jvm_args()
+            assert '-Dhttp.nonProxyHosts=localhost|127.0.0.1|.example.com' in result
+
+    def test_no_proxy_lowercase(self):
+        """Test no_proxy (lowercase) environment variable."""
+        with patch.dict(os.environ, {'no_proxy': 'localhost,127.0.0.1'}, clear=True):
+            result = get_proxy_jvm_args()
+            assert '-Dhttp.nonProxyHosts=localhost|127.0.0.1' in result
+
+    def test_all_proxies_combined(self):
+        """Test with all proxy environment variables set."""
+        env_vars = {
+            'HTTP_PROXY': 'http://user:pass@proxy.example.com:8080',
+            'HTTPS_PROXY': 'https://user:pass@secure-proxy.example.com:8443',
+            'NO_PROXY': 'localhost,127.0.0.1,.local'
+        }
+        with patch.dict(os.environ, env_vars, clear=True):
+            result = get_proxy_jvm_args()
+
+            # HTTP proxy assertions
+            assert '-Dhttp.proxyHost=proxy.example.com' in result
+            assert '-Dhttp.proxyPort=8080' in result
+            assert '-Dhttp.proxyUser=user' in result
+            assert '-Dhttp.proxyPassword=pass' in result
+
+            # HTTPS proxy assertions
+            assert '-Dhttps.proxyHost=secure-proxy.example.com' in result
+            assert '-Dhttps.proxyPort=8443' in result
+            assert '-Dhttps.proxyUser=user' in result
+            assert '-Dhttps.proxyPassword=pass' in result
+
+            # NO_PROXY assertions
+            assert '-Dhttp.nonProxyHosts=localhost|127.0.0.1|.local' in result
+
+    def test_uppercase_takes_precedence(self):
+        """Test that uppercase environment variables take precedence over lowercase."""
+        env_vars = {
+            'HTTP_PROXY': 'http://upper.example.com:8080',
+            'http_proxy': 'http://lower.example.com:3128'
+        }
+        with patch.dict(os.environ, env_vars, clear=True):
+            result = get_proxy_jvm_args()
+            assert '-Dhttp.proxyHost=upper.example.com' in result
+            assert '-Dhttp.proxyPort=8080' in result
+
+    def test_proxy_without_scheme(self):
+        """Test proxy URL without scheme (should add default scheme)."""
+        with patch.dict(os.environ, {'HTTP_PROXY': 'proxy.example.com:8080'}, clear=True):
```

Comment:
Supporting also proxy without scheme prefix (http:// or https:// based on variable it is read from).